src/status_im/bots/events.cljs

```diff
@@ -0,0 +1,98 @@
+(ns status-im.bots.events
+  (:require [re-frame.core :refer [trim-v]]
+            [status-im.utils.handlers :refer [register-handler-db register-handler-fx]]))
+
+;;;; Helper fns
+
+(defn- chats-with-bot [chats bot]
+  (reduce (fn [acc [_ {:keys [chat-id contacts]}]]
+            (let [contacts (map :identity contacts)]
+              (if (some #{bot} contacts)
+                (conj acc chat-id)
+                acc)))
+          []
+          chats))
+
+(defn- subscription-values [subscriptions current-bot-db]
+  (reduce (fn [sub-values [sub-name sub-path]]
+            (assoc sub-values sub-name (get-in current-bot-db sub-path)))
+          {}
+          subscriptions))
+
+;; TODO(janherich): optimze this, for sure we don't need to re-calculate all bot subscriptions every time something in bot db changes
+(defn- check-subscriptions-fx
+  [db {:keys [bot path value]}]
+  (let [{:keys [bot-db chats]} db
+        subscriptions          (get-in db [:bot-subscriptions path])]
+    {:call-jail-function-n
+     (for [{:keys [bot subscriptions name]} subscriptions
+           :let [subs-values (subscription-values subscriptions (get bot-db bot))]]
+       {:chat-id                 bot
+        :function                :subscription
+        :parameters              {:name          name
+                                  :subscriptions subs-values}
+        :callback-events-creator (fn [jail-response]
+                                   (into [[::calculated-subscription
+                                           {:bot    bot
+                                            :path   [name]
+                                            :result jail-response}]]
+                                         (map (fn [chat-id]
```

Comment:
I see that this code is not new in the PR, but i'm wondering why we are dispatching  `::calculated-subscription` for other jails than `bot`? @alwx?

Reply:
@rasom I already had discussion with @alwx about it last week, I was wondering the same thing, this was my question:

Isn’t subscription something which is ultimately tied 1<->1 to particular chat ? Why when some data modifies the subscriptions, it needs to be propagated for each chat where the bot is a contact ? [piece of code where it's done](https://github.com/status-im/status-react/blob/develop/src/status_im/bots/handlers.cljs#L34-#L38)

Let’s for example talk about the `/send` command. The command is global, which means (correct me if I’m wrong), that it’s automatically available in each chat (even if command “registrator”, wallet bot in this case, is not in chat participants).
Let’s say *Alice* has 1-1 chats with *Eve* and *Tom*. Now global commands need to have subscriptions for each chat, which means that we need `:bot-db` and `:bot-subscriptions` paths in app-db for both of those 1-1 chats.
In *Alice*-*Eve* chat, *Alice* types `/send` command, chooses *Eve* as the receiver and enters some amount, moves the slider (something which triggers subscription recalculation) *not yet* submitting the command. 
Now as we get recalculation results from jail, we are not only setting the recalculated data in `:bot-db` for *Alice*-*Eve* chat, but we are trying to find all the chats, where bot is present in contacts.
Now this is very confusing but I figured out that on *Alice’s* device, *Alice*-*Eve* chat id is just *Eve’s* id, so `:bot-db` for it is just something as “command staging place” related to *Eve* and what [this code is trying to achieve](https://github.com/status-im/status-react/blob/develop/src/status_im/bots/handlers.cljs#L34-#L38) is to just find all chats where *Eve* is in participants (for example group chat where *Eve* is also included), so the command context (subscriptions state, etc) remains constant for one recipient. 
However, that doesn’t seem to be the case currently (I just tried it) and to be honest, in can’t really work as it’s implemented.
So is my understanding correct and when yes, do we really need to propagate those subscription changes ?

Reply:
Currently we store calculated subscriptions in map like
```clojure
{:bot-db
 {:bot-id
  {[:path] :data}}}
```
so i suppose if we will change this to 
```clojure
{:bot-db
 {:bot-id
  {:chat-id
   {[:path] :data}}}}
```
we will not need to use this weird propagation. Am i missing something?